app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingRunnable.java

```diff
@@ -263,48 +263,43 @@ private boolean useWakeLock() {
-                        for (String syncthingPID : syncthingPIDs) {
-                            // Set best-effort, low priority using ionice.
-                            niceOut.writeBytes("ionice " + syncthingPID + " be 7\n");
-                        }
-                        niceOut.writeBytes("exit\n");
-                        log(nice.getErrorStream(), Log.WARN, false);
-                        niceOut.flush();
-                        ret = nice.waitFor();
-                        Log.i(TAG_NICE, "ionice performed on " + Constants.FILENAME_SYNCTHING_BINARY);
-                    }
-                } catch (IOException | InterruptedException e) {
-                    Log.e(TAG_NICE, "Failed to execute ionice binary", e);
-                } finally {
-                    try {
-                        if (niceOut != null) {
-                            niceOut.close();
-                        }
-                    } catch (IOException e) {
-                        Log.w(TAG_NICE, "Failed to close shell stream", e);
-                    }
-                    if (nice != null) {
-                        nice.destroy();
-                    }
-                    if (ret != 0) {
-                        Log.e(TAG_NICE, "Failed to set ionice " + Integer.toString(ret));
-                    }
+        Process nice = null;
+        DataOutputStream niceOut = null;
+        int ret = 1;
+        try {
+            List<String> syncthingPIDs = getSyncthingPIDs();
+            if (syncthingPIDs.isEmpty()) {
+                Log.w(TAG, "niceSyncthing: Found no running instances of " + Constants.FILENAME_SYNCTHING_BINARY);
+            } else {
+                nice = Runtime.getRuntime().exec((mUseRoot) ? "su" : "sh");
+                niceOut = new DataOutputStream(nice.getOutputStream());
+                for (String syncthingPID : syncthingPIDs) {
+                    // Set best-effort, low priority using ionice.
+                    niceOut.writeBytes("/system/bin/ionice " + syncthingPID + " be 7\n");
                 }
+                niceOut.writeBytes("exit\n");
+                log(nice.getErrorStream(), Log.WARN, false);
+                niceOut.flush();
+                ret = nice.waitFor();
+                Log.i(TAG_NICE, "ionice performed on " + Constants.FILENAME_SYNCTHING_BINARY);
             }
-        }.start();
+        } catch (IOException | InterruptedException e) {
+            Log.e(TAG_NICE, "Failed to execute ionice binary", e);
+        } finally {
+            try {
+                if (niceOut != null) {
+                    niceOut.close();
+                }
+            } catch (IOException e) {
+                Log.w(TAG_NICE, "Failed to close shell stream", e);
+            }
+            if (nice != null) {
+                nice.destroy();
+            }
+            if (ret != 0) {
+                Log.e(TAG_NICE, "Failed to set ionice " + Integer.toString(ret));
+            }
+        }
```

Comment:
Why was this in a thread before?

Reply:
Removing it was just a proposal as I think the person getting this in meant to address an older syncthing version before 0.14.47. I'm reverting this as I agree with you we can remove it later. To sum up my answer to the question: No need for a new thread. Can stay as it is and won't do harm.
Revert with commit https://github.com/syncthing/syncthing-android/pull/1119/commits/bd08037e07e81c37d54cca5a4e6e30ba3d62e880

Reply:
`"nices out itself" on since binary version 0.14.47+`, not ionice, which is what this is doing.